packages/loopring_v3/contracts/amm/LoopringAmmPool.sol

```diff
@@ -140,10 +183,29 @@ contract LoopringAmmPool is
         state.withdrawWhenOffline();
     }
 
-    function updateExchangeOwnerAndFeeBips()
+    function depositAssetsToL2(
+        uint96[] memory amounts
+        )
+        external
+        nonReentrant
+        onlyWhenOnline
+        onlyFromExchangeOwner
+    {
+        for(uint i = 0; i < state.tokens.length; i++) {
+            require(amounts[i] != 0, "INVALID_DEPOSIT_AMOUNT");
+            state.deposit(state.tokens[i].addr, amounts[i]);
+        }
+    }
+
+    function transferOut(
+        address to,
```

Comment:
Would be fine with me, though I guess it could be possible there are different "vault" contracts the asset manager manages depending on token or something

>  I don't think the manager should be an EOA address.

True, but just checking that the asset manager is a contract doesn't really have any security benefits I think because contracts are not necessarily more secure than a EOA.